rust/src/delta_arrow.rs

```diff
@@ -369,12 +372,24 @@ pub(crate) fn delta_log_schema_for_table(
         ));
     }
 
+    // create remove fields with or without extendedFileMetadata
+    let mut remove_fields = REMOVE_FIELDS.clone();
+    if use_extended_remove_schema {
+        remove_fields.extend(REMOVE_EXTENDED_FILE_METADATA_FIELDS.clone());
+    }
+
```

Comment:
@xianwill btw, wouldn't the easier hotfix will be a just write a extended_file_metadata=false without any other columns

Reply:
however for the full compitability with delta 1.0 I agree that we must include them

Reply:
@mosyp - Just setting `extended_file_metadata` to false doesn't fix the issue. Same error still happens. If its false, the schema for the other three must be omitted entirely.